Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs

```diff
@@ -30,5 +35,67 @@ public void DiscordApiCredentialsValidator_Test()
                                                           keyValuePairs,
                                                           ref resultLevelKind);
         }
+
+        [Fact]
+        public void DiscordCredentialsValidator_MockHttpTests()
+        {
+            var testCases = new[]
+            {
+                new
+                {
+                    Title = "Testing Valid Credentials",
+                    HttpStatusCode = HttpStatusCode.OK,
+                    ExpectedValidationState = ValidationState.Authorized,
+                    ExpectedMessage = "The compromised asset is 'a'.",
+                    HttpContent = (HttpContent)null,
+                },
+                new
+                {
+                    Title = "Testing Invalid Credentials",
+                    HttpStatusCode = HttpStatusCode.Unauthorized,
+                    ExpectedValidationState = ValidationState.Unauthorized,
+                    ExpectedMessage = "The provided secret is not authorized to access 'a'.",
+                    HttpContent = (HttpContent)null,
+                },
+                new
+                {
+                    Title = "Testing Unknown Status code",
+                    HttpStatusCode = HttpStatusCode.NotFound,
+                    ExpectedValidationState = ValidationState.Unknown,
+                    ExpectedMessage = "An unexpected HTTP response code was received: 'NotFound'.",
+                    HttpContent = (HttpContent)null,
+                },
+            };
+            const string fingerprintText = "[id=a][secret=b]";
+
+            var sb = new StringBuilder();
+            foreach (var testCase in testCases)
+            {
+                string message = string.Empty;
+                ResultLevelKind resultLevelKind = default;
+                var fingerprint = new Fingerprint(fingerprintText);
+                var keyValuePairs = new Dictionary<string, string>();
+
+                MockHelper.ResetStaticInstance<SquareCredentialsValidator>();
```

Comment:
>SquareCredentialsValidator [](http://example.com/codeflow?start=47&length=26)

this should be DiscordApiCredentialsValidator

Reply:
fixed